### ISSUE TITLE
swupdate_get_sha256 currently does not search for files in artifactory

### ISSUE THREAD

**Issue by Sperrus** [sbabic/swugenerator#10]
The sw-description function swupdate_get_sha256 currently does not search for the filename in artifactory. 
Meanwhile swupdate_get_size does. This is somewhat inconsistent.

**Comment by Sperrus** [comment]
Hmmm, I just noticed this function is completely obsolete as process_entry always adds an sha256 to new entries.
This is not really ideal, when the target swupdate does not have CONFIG_SIGNED_IMAGES enabled as this always throws an error

**Comment by sbabic** [comment]
I will check, but obsolete function can be dropped. It is not required to enable CONFIG_SIGNED_IMAGE, just CONFIG_HASH_VERIFY (it is always suggested to set it on) must be set.

**Comment by Sperrus** [comment]
You are correct, CONFIG_HASH_VERIFY should always be set. 
But it is possible to configure swupdate without this feature and therefore swugenerator should be able to generate valid images for this configuration.

**Comment by sbabic** [comment]
Yes, confirmed, this is a limitation in swugenerator - a flag to disable sha256 should be added.

**Comment by Sperrus** [comment]
I am not sure if a new flag is necessary. 
Imho the sw-description file should be the deciding factor on what is being done. If there is no sha field in sw-description, no sha should be added for that entry.

**Comment by sbabic** [comment]
sw-description is the master, but the generator should help in cases this is mandatory.  If sha256 must be added, a missing sha256 in case of signed image will result in runtime error (this is the case when built in Yocto).  The only drawback is for the use case where CONFIG_HASH_VERIFY is off (probably your use case), but this is a minority of projects. So I think it is better to tell the generator to disable the computation of the hash, but let it to create in the other cases.
